govcd/externalnetwork.go

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+	"net/http"
+)
+
+type ExternalNetwork struct {
+	ExternalNetwork *types.ExternalNetwork
+	client          *Client
+}
+
+func NewExternalNetwork(cli *Client) *ExternalNetwork {
+	return &ExternalNetwork{
+		ExternalNetwork: new(types.ExternalNetwork),
+		client:          cli,
+	}
+}
+
+func getExternalNetworkHref(client *Client) (string, error) {
+	extensions, err := getExtension(client)
+	if err != nil {
+		return "", err
+	}
+
+	for _, extensionLink := range extensions.Link {
+		if extensionLink.Type == "application/vnd.vmware.admin.vmwExternalNetworkReferences+xml" {
+			return extensionLink.HREF, nil
+		}
+	}
+
+	return "", errors.New("external network link isn't found")
+}
+
+func (externalNetwork ExternalNetwork) GetByName(networkName string) error {
```

Comment:
I am puzzled by this method. We have an entity that looks for an instance of the same type, and then returns an error.
Semantically, I can't figure out how this function fits with the rest of the library. I would expect to see a function in `system.go`, using a `VCDClient` as an accessor, and returning an external network.

What happens exactly here, instead? When the function is successful, what do I get? The original external network (the one used for finding by name) becomes the found one?

Reply:
This was Megalord idea, I understand it after using it, but I agree this isn't very clean approach. I will look for idea to improve it.

Reply:
This is how other may think about current structure of API, we have `externalNetwork` which consist of `client` and `externalNetwork` which is types struct. What Megalord did was - create `externalNetwork` with client  and fetch `types.externalNetwork` then with 'getByName'

Reply:
I like a `Get` function to return something other than an error. 
A function that returns an error is one that runs a procedure, and informs on the outcome.
I am still unsure I understand what this function does. What happens to the external network being found? and what changes in the calling object?

If it searches for something, it should return what it finds, and even then, I don't like to use an object to find an entity of the same type, and potentially itself. 
If the function changes something in the calling object, the name should say what this operation was.

Reply:
Yep, I agree. I changed to be more understandable